src/base/bittorrent/session.h

```diff
@@ -798,6 +803,8 @@ namespace BitTorrent
 
         QList<MoveStorageJob> m_moveStorageQueue;
 
+        QHostAddress m_lastExternalIP;
```

Comment:
Seems you don't really need the facilities of QHostAddress, you should use QString instead.

Reply:
Already change m_lastExternalIP into QString, and remove include<QHostname> from session.h.